### PR TITLE
Add feature for person, students and rental

### PR DESCRIPTION
**Pull request by Caren-Koroeny**
<h1> Hello There 👋🏻 </h1>

> In this PR, I:

**Created class Person with the following:**
  
- [ ]  Instance vars: @id, @name, and @Age.
- [ ] Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have
- [ ] default values of "Unknown" and true.
- [ ] Getters for @id, @name, and @Age.
- [ ] Setters for @name and @Age.
- [ ] Private method is_of_age? that returns true if @Age is greater or equal to 18 and false otherwise.
- [ ] Public method can_use_services? that returns true if person is of age or if they have permission from parents.

**Created class Student with the following:**

- [ ] Inherits from Person.
- [ ] Constructor extends parent's constructor by adding @classroom and a parameter for it.
- [ ] Method play_hooky that returns "¯(ツ)/¯".

**Created class Teacher with the following:**

- [ ] Inherits from `Person`.
- [ ] Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
- [ ] Override can_use_services? so it always returns true.

<h1>Each class was saved in a separate file.</h1>

<h1> Kindly Review .... Happy Reviewing </h1>

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative './person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(_id, age, classroom, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
```
Comment (TedLivist) [inline student.rb]: - [ ] Kindly remove `_id` from the list of parameters here as it auto-generated in the Person class so we don't need to pass it in.

- [ ] In the `super` parameters kindly re-arrange them, `age` should come before `name` based on how the Person constructor is set up.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require_relative './person'
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(id, name, age, parent_permission)
```
Comment (TedLivist) [inline teacher.rb]: - [ ] Kindly remove `id` from the list of parameters here.

- [ ] In the `super` parameters kindly re-arrange them, `age` should come before `name` based on how the Person constructor is set up. Also ensure the `parent_permission` is assigned properly.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Caren-Koroeny,

Good job so far!
✔️ Classes are in different files based on Ruby conventions
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative './person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(_age, classroom, name = 'unknown', parent_permission: true)
+    super(nage, name, parent_permission: parent_permission)
```
Comment (nahnahdev) [inline student.rb]: - [ OPTIONAL ] It looks like you made a typo here in the super parameters. You have `nage` instead of `age`. Kindly fix this 👍

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Caren-Koroeny , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented 🔥 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Caren-Koroeny , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented 🔥 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**